File: src/main/java/io/sailex/aiNpc/client/model/NPC.java
package io.sailex.aiNpc.client.model;

import io.sailex.aiNpc.client.llm.ILLMClient;
import io.sailex.aiNpc.client.npc.NPCContextGenerator;
import io.sailex.aiNpc.client.npc.NPCController;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import net.minecraft.client.network.ClientPlayerEntity;

@Data
@AllArgsConstructor
public class NPC {

	private UUID id;
	private ClientPlayerEntity npcEntity;
	private NPCController npcController;
	private NPCContextGenerator npcContextGenerator;
	private ILLMClient llmService;
}

File: ai-npc-client/src/main/java/io/sailex/aiNpc/client/model/NPC.kt
package io.sailex.aiNpc.client.model

import io.sailex.aiNpc.client.llm.ILLMClient
import io.sailex.aiNpc.client.npc.NPCContextGenerator
import io.sailex.aiNpc.client.npc.NPCController
import net.minecraft.client.network.ClientPlayerEntity
import java.util.*

/**
 * Represents an NPC in the game and its llm service, context generator, and controller
 */
class NPC(
    val id: UUID,
    val npcEntity: ClientPlayerEntity,
    val npcController: NPCController,
    val npcContextGenerator: NPCContextGenerator,
    val llmService: ILLMClient
)
